app/src/main/java/com/tistory/dividendcalendar/presentation/notice/NoticeFragment.kt

```diff
@@ -0,0 +1,162 @@
+                requireActivity().packageManager.getPackageInfo(requireActivity().packageName, 0)
+            val versionCode = info.versionCode
+            val versionInfo = if (versionCode == versionCodeFromServer) {
+                "(${getString(R.string.latest_version)})"
+            } else {
+                "(${getString(R.string.need_update)})"
+            }
+
+            val versionName = "ver. ${info.versionName} $versionInfo"
+
+            requireContext().alert(title = versionName, message = getString(R.string.app_info)) {
+                positiveButton(getString(R.string.question)) {
+                    sendEmail()
+                }
+                cancelButton {
+                    //..
+                }
+            }.show()
+        }
+    }
+
+    private fun sendEmail() {
+        val email = Intent(Intent.ACTION_SEND).apply {
+            type = "plain/Text"
+            val address = arrayOf("dlwls5201@gmail.com")
+            putExtra(Intent.EXTRA_EMAIL, address)
+            putExtra(Intent.EXTRA_SUBJECT, "<" + getString(R.string.app_name) + ">")
+            putExtra(
+                Intent.EXTRA_TEXT,
+                "AppVersion :${BuildConfig.VERSION_NAME}\nDevice : ${Build.MODEL}\nAndroid OS : ${Build.VERSION.SDK_INT}\n\n Content :\n"
+            )
+        }
+        startActivity(email)
+    }
+
+    private var versionCodeFromServer = -1
+
+    private fun loadData() {
+        remoteConfig.fetchAndActivate()
+            .addOnCompleteListener(requireActivity()) { task ->
+                if (task.isSuccessful) {
+                    val updated = task.result
+                    Dlog.d("Config params updated: $updated")
+
+                    val version = remoteConfig.getString("version")
+                    versionCodeFromServer = version.toIntOrNull() ?: -1
+                    Dlog.d("versionCodeFromServer : $versionCodeFromServer")
+
+                    val noticeItemsJsonString = remoteConfig.getString("notice_items")
+                    Dlog.d("noticeItemsJsonString : $noticeItemsJsonString")
+
+                    try {
+                        val noticeResponse =
+                            convertJsonArrayToList<NoticeResponse>(noticeItemsJsonString)
+
+                        val items = mutableListOf<NoticeItem>()
+
+                        findNoticeForDisplayLanguage(noticeResponse).items.mapTo(items) { it.mapToItem() }
+
+                        noticeAdapter.replaceAll(items)
+                    } catch (e: Exception) {
+                        Dlog.d("Parsing error : ${e.message}")
+                    }
+                } else {
+                    Dlog.d("Fetch failed")
+                }
+            }
+    }
+
+    private val gson = Gson()
+
+    private inline fun <reified T> convertJsonArrayToList(jsonString: String): List<T> {
+        //return gson.fromJson(jsonString, object : TypeToken<ArrayList<T>>() {}.type)
+        return gson.fromJson(
```

Comment:
```suggestion
        return Gson().fromJson(
```
변수 선언 없이 바로 사용할 수 있을 것 같아 제안드립니다~

Reply:
현재는 함수를 재사용하고 있지 않아 위처럼 사용해도 문제없을것 같아요! 하지만 Gson().fromJson 형식으로 사용하게 되면 함수를 호출할 때마다 불필요하게 Gson() 객체를 매번 생성해야 될것 같아서 따로 변수로 선언 후 재사용하기 위해 위처럼 작업했습니다.